aws/resource_aws_launch_template_test.go

```diff
@@ -1077,6 +1125,80 @@ resource "aws_launch_template" "test" {
 }
 `
 
+const testAccAWSLaunchTemplateConfig_associatePublicIpAddressTrue = `
+resource "aws_vpc" "test" {
+  cidr_block = "10.1.0.0/16"
+}
+
+resource "aws_subnet" "test" {
+  vpc_id = "${aws_vpc.test.id}"
+  cidr_block = "10.1.0.0/24"
+}
+
+resource "aws_network_interface" "test" {
+  subnet_id = "${aws_subnet.test.id}"
+}
+
+resource "aws_launch_template" "test" {
+  name = "network-interface-launch-template"
```

Comment:
This name needs to be randomized to prevent issues like:

```
--- FAIL: TestAccAWSLaunchTemplate_networkInterface (23.50s)
    testing.go:635: Step 0 error: errors during apply:

        Error: InvalidLaunchTemplateName.AlreadyExistsException: Launch template name already in use.
```

Combining with the suggested update above:

```go
func testAccAWSLaunchTemplateConfig_associatePublicIpAddress(rName, associatePublicIpAddress string) string {
  return fmt.Sprintf(`
# ...
resource "aws_launch_template" "test" {
  name = %[1]q

  network_interfaces {
    associate_public_ip_address = %[2]s
    network_interface_id        = "${aws_network_interface.test.id}"
    ipv4_address_count          = 2
  }
}
`, rName, associatePublicIpAddress)
```